### DISCUSSION TITLE
Launch command on remote ssh session

### DISCUSSION THREAD

**Discussion by AprilArcus** [kovidgoyal/kitty#5987] [General]
If I execute `kitty +kitten ssh my-remote` and then `kitty @ launch --cwd=current` from the resulting SSH session, I see a new kitty window connected to `my-remote`.

However, if I pass a command as an argument, e.g. `kitty @ launch --cwd=current top`, the command is executed on my local environment, not my `my-remote`. How can I use `kitty @ launch` to open a new command after establishing an SSH connection and changing to the appropriate directory?

**Comment by AprilArcus** [comment]
I experimented a little bit and found that this accomplishes what I want:

`kitty @ launch kitty +kitten ssh -t "$(whoami)@$(hostname)" "cd $(pwd); my_command"`

I would propose that

`kitty @ launch --cwd=current my_command`

could be shorthand for this.

**Comment by kovidgoyal** [comment]
IIRC currently kitty only forwards over ssh if you are running the shell,
this is because often when running a program it might be one that you
want the local version of, for example when browsing scrollback or
running a kitten. I suppose I could have requests received over remote
control from a remote system default to running the specified command
remotely, or maybe add a flag to launch to control the behavior.

**Comment by kovidgoyal** [comment]
There you go: https://github.com/kovidgoyal/kitty/commit/7c8c7fe3a25118bc26be4887d111ae1b10839a38
